crates/sui-adapter-transactional-tests/tests/dynamic_fields/receive_remove_add_back_and_remove_type.move

```diff
@@ -0,0 +1,111 @@
+// they won't show up in the child_object_effects in the object runtiem. We must look at either transfers
+// or deleted_object_ids to figure them out.
+
+//# init --addresses test=0x0 --accounts A
+
+//# publish
+module test::m1 {
+    use sui::object::{Self, UID};
+    use sui::transfer::{Self, Receiving};
+    use sui::tx_context::{Self, TxContext};
+
+    struct Object has key, store {
+        id: UID,
+    }
+
+    struct C1 has key, store {
+        id: UID,
+    }
+
+    struct C2 has key, store {
+        id: UID,
+    }
+
+    struct Wrapper<T: store> has key, store {
+        id: UID, 
+        value: T,
+    }
+
+    public entry fun create(ctx: &mut TxContext) {
+        let o = Object { id: object::new(ctx) };
+        let c1 = C1 { id: object::new(ctx) };
+        let c2 = C2 { id: object::new(ctx) };
+        let o_address = object::id_address(&o);
+        transfer::public_transfer(o, tx_context::sender(ctx));
+        transfer::public_transfer(c1, o_address);
+        transfer::public_transfer(c2, o_address);
+    }
+
+    public entry fun test_dof(parent: &mut Object, c1: Receiving<C1>, c2: Receiving<C2>) {
+        let c1 = transfer::receive(&mut parent.id, c1);
+        sui::dynamic_object_field::add(&mut parent.id, 0, c1);
+        let c1: C1 = sui::dynamic_object_field::remove(&mut parent.id, 0);
+        transfer::public_transfer(c1, @test);
+
+        let c2 = transfer::receive(&mut parent.id, c2);
+        sui::dynamic_object_field::add(&mut parent.id, 0, c2);
+        let C2 { id } = sui::dynamic_object_field::remove(&mut parent.id, 0);
+        object::delete(id);
+    }
+
+    public entry fun test_df(parent: &mut Object, c1: Receiving<C1>, c2: Receiving<C2>) {
+        let c1 = transfer::receive(&mut parent.id, c1);
+        sui::dynamic_field::add(&mut parent.id, 0, c1);
+        let c1: C1 = sui::dynamic_field::remove(&mut parent.id, 0);
+        transfer::public_transfer(c1, @test);
+
+        let c2 = transfer::receive(&mut parent.id, c2);
+        sui::dynamic_field::add(&mut parent.id, 0, c2);
+        let C2 { id } = sui::dynamic_field::remove(&mut parent.id, 0);
+        object::delete(id);
+    }
+
+    // Try to "wash" the receiving object through a dynamic object field and then wrap it in a wrapper object.
+    public entry fun test_dof_wrapper(parent: &mut Object, c1: Receiving<C1>, c2: Receiving<C2>, ctx: &mut TxContext) {
+        let c1 = transfer::receive(&mut parent.id, c1);
+        sui::dynamic_object_field::add(&mut parent.id, 0, c1);
+        let c1: C1 = sui::dynamic_object_field::remove(&mut parent.id, 0);
+        let w = Wrapper { id: object::new(ctx), value: c1 };
+        transfer::public_transfer(w, @test);
+
+        let c2 = transfer::receive(&mut parent.id, c2);
+        sui::dynamic_object_field::add(&mut parent.id, 0, c2);
+        let C2 { id } = sui::dynamic_object_field::remove(&mut parent.id, 0);
+        object::delete(id);
```

Comment:
Is this doing anything different than the test above?

Reply:
Not really, I'll update it though to do some more wrapping which should test to make sure we track wrappings through this.